src/molecules/Site.h

```diff
@@ -126,7 +130,17 @@ class LJcenter : public Site {
 		Site::readXML(xmlconfig);
 		xmlconfig.getNodeValueReduced("epsilon", _epsilon);
 		xmlconfig.getNodeValueReduced("sigma", _sigma);
-		xmlconfig.getNodeValue("shifted", _shiftRequested);
+		// Read shifted value and check if its boolean (true/false) to avoid legacy-errors
+		std::string strShifted = "false";
+		xmlconfig.getNodeValue("shifted", strShifted);
+		// Convert to lower case to avoid input errors
+		std::transform(strShifted.begin(), strShifted.end(), strShifted.begin(), [](auto c) { return std::tolower(c); });
+		if (strShifted == "true" || strShifted == "false") {
+			std::istringstream(strShifted) >> std::boolalpha >> _shiftRequested;
+		} else {
+			global_log->error() << "Parameter <shifted> of components has to be either set to 'true' or 'false'" << std::endl;
+			mardyn_exit(1);
+		}
 	}
```

Comment:
This should not be necessary at as the xml parser code already handles "YES", "TRUE", "ON" ignoring case. See https://github.com/ls1mardyn/ls1-mardyn/blob/b6f274e625fe7a8c07861f09c2b6c9c5e0243f2e/src/utils/xmlfile.cpp#L589

Reply:
The point of this code is to check if the input value of `<shifted>` is really a boolean as it was a float before introducing the `_shiftRequested` variable a few month ago. If we would just do `xmlconfig.getNodeValue("shifted", _shiftRequested);` then a float will be probably converted to false. Even though a user would expect a number != 0 to shift the potential. This code makes sure that an old version of the components where the shift value was set to be `0.097901347` (shift for cutoff radius =2.5) can not be used without an error. If we just parsed this value, the potential would not be shifted and the simulation run would yield wrong results.

Reply:
This is [now](https://github.com/ls1mardyn/ls1-mardyn/pull/255/commits/981cddfbf729484d9b2e30adb4019de07b166076) solved by checking all boolean input values, and [additionally](https://github.com/ls1mardyn/ls1-mardyn/pull/255/commits/0f8092e6329dc537d166008ec378c6ad7af36709) also other user input via the xmlfile.